### PR TITLE
Fetch workspace from Master if all nodes are offline

### PR DESCRIPTION
**Pull request by alokjoshi89**
This PR adds a feature to fetch workspace from Jenkins Master, if all nodes are offline.

### CODE REVIEW

**Comment by alokjoshi89** [comment]
Can we get this merged :)

**Comment by iocanel** [comment]
Some of the requested changes in the previous pull request were not addressed. For sure the `redundant null checks` haven't. I am not sure about the changes @carlossg requested.

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesWorkspaceBrowser.java [hunk 1]
```diff
@@ -21,20 +21,25 @@
...
-                List<PodTemplate> podTemplates = kubernetesCloud.getTemplates();
-                for(PodTemplate podTemplate : podTemplates) {
-                    if(ObjectUtils.equals(podTemplate.getLabel(), assignedLabel)) {
-                        String workspacePath = Jenkins.getInstance().getWorkspaceFor((TopLevelItem) job).toString();
-                        LOGGER.info("Workspace Path: " + workspacePath);
-                        File workspace = new File(workspacePath);
-                        LOGGER.info("Workspace exists ? " + workspace.exists());
-                        if (workspace.exists()) {
-                            return new FilePath(workspace);
+        if(job != null) {
```
Comment (iocanel) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesWorkspaceBrowser.java]: You don't need to check if job is null. A null job will never go through `job instanceof AbstractProject`.

**Comment by iocanel** [comment]
Ohh I am sorry, I was looking at the commits one by one and didn't realize that you address my concern in a later comment.

**Comment by alokjoshi89** [comment]
@carlossg can you please comment on the PR and approve if its looks good?

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesWorkspaceBrowser.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                List<KubernetesCloud> kubernetesClouds = jenkinsInstance.clouds.getAll(KubernetesCloud.class);
+
+                //We only care about Kubernetes clouds for the Jenkins instance.
+                //Go through the list of Kube clouds and check if we find a match for the job label with any of the pod templates
+                for (KubernetesCloud kubernetesCloud : kubernetesClouds) {
+                    if (kubernetesCloud != null) {
+                        List<PodTemplate> podTemplates = kubernetesCloud.getTemplates();
+                        for (PodTemplate podTemplate : podTemplates) {
+                            if (ObjectUtils.equals(podTemplate.getLabel(), assignedLabel)) {
+                                FilePath filePath = Jenkins.getInstance().getWorkspaceFor((TopLevelItem) job);
```
Comment (carlossg) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesWorkspaceBrowser.java]: you already have `jenkinsInstance`

**Comment by gytisgreitai** [comment]
@alokjoshi89  do you intend to finish this up?

**Comment by farahfa** [comment]
So is this not gonna happen? It would be very useful to debug jobs quickly by being able to access the workspace after the slave node is gone.

**Comment by yrsurya** [comment]
is this PR merged already and available if so please anyone can guide how to get workspace after agent removed.

**Pull request closed without merging**